### PR TITLE
Multi cluster

### LINKED ISSUE #98: Investigate kind
We support Minikube, EKS, and GKE -- what about kind?

### LINKED ISSUE #118: Multi cluster instrumentation
Currently valet uses only uses the users main kubeconfig file, usually `~/.kube/config` or the one specified by `KUBECONFIG` in the parent context. And within this kube config, it uses the context set by the cluster creation command.

This results in valet runs that only support single cluster workflows. In order to expand to multi cluster workflows I see a couple paths forward:

1) Allow for multi cluster instrumentation simultaneously by allowing the kubeconfig/context to be configurable, and making those values required for any kube related operations.

2) Allow for easier running of valet in an isolated docker environment. So 2 configs can be run simultaneously in seperate environments. The registries introduced in `0.4.0` should make this much easier. However, this would limit the ability to have multi cluster workflows, by limiting each workflow to one cluster in the env.

### LINKED ISSUE #132: Helm v3 and Kube 1.16 to support new go-utils
the latest go-utils upgraded to helm v3, and kube 1.16. Need to bump to support that

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/valet/issues/132
resolves https://github.com/solo-io/valet/issues/98
resolves https://github.com/solo-io/valet/issues/118

### CODE REVIEW

File: cli/cmd/config/global.go [hunk 1]
```diff
@@ -104,18 +105,18 @@ func LoadGlobalConfig(opts *options.Options) (*ValetGlobalConfig, error) {
...
 	path, err := GetGlobalConfigPath(opts)
 	if err != nil {
-		cmd.Stderr().Println("Could not determine global config path: %s", err.Error())
+		cmd.Stderr(context.TODO()).Printf("Could not determine global config path: %s", err.Error())
 		return err
 	}
 
 	bytes, err := yaml.Marshal(config)
 	if err != nil {
-		cmd.Stderr().Println("Failed to marshal config: %s", err.Error())
```
Comment (rickducott) [inline cli/cmd/config/global.go:113]: can't these all use `opts.Top.Ctx`?

File: cli/cmd/set-context/cmd.go [hunk 2]
```diff
@@ -33,12 +32,16 @@ func setContext(opts *options.Options) error {
...
 		return err
 	}
 	if cfg.Cluster == nil {
 		return MustSpecifyClusterError
 	}
-	return cfg.Cluster.SetContext(opts.Top.Ctx, cmd.DefaultCommandRunner())
+	if err := input.RenderFields(cfg.Cluster); err != nil {
+		return err
+	}
+	// TODO(EItanya): clean up deref
```
Comment (rickducott) [inline cli/cmd/set-context/cmd.go:45]: ?

File: cli/internal/ensure/cmd/commands.go [hunk 3]
```diff
@@ -160,10 +207,21 @@ func (c *Command) logCommand(ctx context.Context) {
...
+		Stdout(ctx).Printf("Running command: %s", command)
+	}
+}
+
+func (c *Command) loadEnv(cmd *exec.Cmd) {
+	if cmd.Env == nil {
+		cmd.Env = os.Environ()
+	}
+	for key, val := range c.Env {
+		// Stdout(ctx).Printf("")
```
Comment (rickducott) [inline cli/internal/ensure/cmd/commands.go:219]: remove

File: cli/internal/ensure/resource/multicluster/ref.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	"context"
+
+	"github.com/solo-io/valet/cli/internal/ensure/resource/workflow"
+	"gopkg.in/yaml.v2"
+
+	"github.com/solo-io/valet/cli/internal/ensure/cmd"
+	"github.com/solo-io/valet/cli/internal/ensure/resource/render"
+)
+
+// A Workflow Ref is a path to a file that can be deserialized into a cluster workflow
```
Comment (rickducott) [inline cli/internal/ensure/resource/multicluster/ref.go:13]:
```suggestion
// A Multicluster Ref is a path to a file that can be deserialized into a cluster workflow
```

File: cli/internal/ensure/resource/multicluster/workflow.go [hunk 5]
```diff
@@ -1,60 +1,69 @@
...
 
 import (
 	"context"
 
 	"github.com/solo-io/valet/cli/internal/ensure/cmd"
 	"github.com/solo-io/valet/cli/internal/ensure/resource/cluster"
 	"github.com/solo-io/valet/cli/internal/ensure/resource/render"
+	"github.com/solo-io/valet/cli/internal/ensure/resource/workflow"
 	"gopkg.in/yaml.v2"
 )
```
Comment (rickducott) [inline cli/internal/ensure/resource/multicluster/workflow.go:12]: Not sure why this was moved, I think there are now two different `Workflow` types in different packages that are roughly the same. Can we consolidate to one? Makes sense to me that `workflow` is a single cluster workflow with a specific cluster, and there may be a separate config for multi, based on the way you've designed the commands

File: cli/cmd/ensure/multiple.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"github.com/solo-io/go-utils/cliutils"
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/valet/cli/cmd/common"
+	"github.com/solo-io/valet/cli/options"
+	"github.com/spf13/cobra"
+)
+
+func Multiple(opts *options.Options, optionsFunc ...cliutils.OptionsFunc) *cobra.Command {
```
Comment (rickducott) [inline cli/cmd/ensure/multiple.go:11]: seems like we use `multi`, `multicluster`, and now `multiple`. can we rename this `multi` or `multicluster`?

**Review by rickducott — COMMENTED** [review]
I'm not sure when you would use `ensure multiple` vs `ensure`. There are a bunch of docs that will be out of date when this goes in, so we should update them -- and the docs updates will probably help clarify what is intended here.

File: cli/cmd/set-context/cmd.go (hunk 2 above)
Comment (EItanya) [reply cli/cmd/set-context/cmd.go:45]: I think I wasn't happy with the *input. I don't really remember though. Will double check

**Pull request closed without merging**